components/brave_new_tab_ui/components/search/config.ts

```diff
@@ -26,38 +22,57 @@ const getConfig = () => {
   return cache!
 }
 
-// Determines whether any search engines are enabled
-export const hasEnabledEngine = () => Object.values(getConfig()).find(c => c)
+export function useNtpWidgetDefaultSearchEngine() {
+  const [lastUsed, setLastUsed] = useNewTabPref('lastUsedNtpSearchEngine')
+  const lastLocalStorage = localStorage.getItem(LAST_SEARCH_ENGINE_KEY)
+  const config = getEnabledSearchEngines()
 
-export const maybeEnableDefaultEngine = () => {
-  // If no engines are enabled, enable the default one.
-  if (!hasEnabledEngine()) {
-    setEngineEnabled({ host: braveSearchHost }, true)
-  }
-}
+  const setLastNtpSearchEngine = React.useCallback((engine: SearchEngineInfo) => {
+    setLastUsed(engine.host);
+    localStorage.setItem(LAST_SEARCH_ENGINE_KEY, engine.host);
+  }, [setLastUsed])
 
-export const setEngineEnabled = (engine: { host: string }, enabled: boolean) => {
-  const config = getConfig()
-  config[engine.host] = enabled
+  const lastNtpSearchEngine = React.useCallback(() => {
+    let last = lastUsed;
 
-  localStorage.setItem(ENABLED_SEARCH_ENGINES_KEY, JSON.stringify(config))
-}
+    if (lastLocalStorage && config[lastLocalStorage] && lastUsed !== lastLocalStorage) {
+      setLastUsed(lastLocalStorage);
+      last = lastLocalStorage;
+    }
 
-export const isSearchEngineEnabled = (engine: SearchEngineInfo) => getConfig()[engine.host]
+    // If the last search engine we used has been disabled or doesn't exist in the config, return the first enabled
+    // one, or Brave Search.
+    if (!last || !config[last]) {
+      return Object.keys(config).find(key => config[key]) ?? braveSearchHost
+    }
 
-export const getDefaultSearchEngine = () => {
-  const config = getConfig()
-  const last = localStorage.getItem(LAST_SEARCH_ENGINE_KEY)
+    return last
+  }, [lastUsed, lastLocalStorage, config]);
 
-  // If the last search engine we used has been disabled, return the first enabled
-  // one, or Brave Search.
-  if (!config[last!]) {
-    return Object.entries(config).find(([key, value]) => value)?.[0]
-      ?? braveSearchHost
+  return {
+    lastNtpSearchEngine: lastNtpSearchEngine(),
```

Comment:
I'm a bit wary of this hook - it won't necessarily be retriggered when the enabled list of search engines changes (which could change the last enabled search engine).

I think this hook should use `useEnabledSearchEngines` rather than `getConfig` to ensure it gets rerun when `enabledSearchEngines` are changed.